core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java

```diff
@@ -137,33 +136,60 @@ public static InputStream getTumorTypeInputStreamFromGitHub(Version version) {
         return tumorTypes;
     }
 
-    private static Map<String, TumorType> parseFromRaw(InputStream inputStream, Version version) {
-        TsvParserSettings settings = new TsvParserSettings();
+    private static Map<String, TumorType> loadFromRepository(Version version) {
+        List<OncoTreeNode> oncoTreeNodes = oncoTreeRepository.getOncoTree();
+        Map<String, TumorType> tree = new HashMap<>();
 
-        //the line separator sequence is defined here to ensure systems such as MacOS and Windows
-        //are able to process this file correctly (MacOS uses '\r'; and Windows uses '\r\n').
-        settings.getFormat().setLineSeparator("\n");
-
-        // creates a TSV parser
-        TsvParser parser = new TsvParser(settings);
-
-        Map<String, TumorType> tumorTypes = new HashMap<>();
+        TumorType root = new TumorType();
+        root.setCode("TISSUE");
+        root.setName("Tissue");
+        // tree seems to only contain root node
+        tree.put(root.getCode(), root);
 
-        List<String[]> allRows = parser.parseAll(inputStream);
+        Map<String, TumorType> allNodes = new HashMap<>();
 
-        TumorType tumorType = new TumorType();
-        tumorType.setCode("TISSUE");
-        tumorType.setName("Tissue");
+        for (OncoTreeNode node : oncoTreeNodes) {
+            logger.debug("OncoTreeNode: code='" + node.getCode() + "', name='" + node.getName() + "'");
+            TumorType tumorType = initTumorType(node, version);
+            allNodes.put(tumorType.getCode(), tumorType);
+        }
 
-        //Iterate each row and assign tumor type to parent following the order of appearing
-        for (String[] row : allRows.subList(1, allRows.size())) {
-            tumorType.setChildren(attachTumorType(tumorType.getChildren(), row, 0, version, "TISSUE"));
+        // now we have all nodes, fill in children
+        for (TumorType tumorType : allNodes.values()) {
+            // TISSUE is root and has no parent, skip
+            if (!tumorType.getCode().equals("TISSUE")) {
+                if (tumorType.getParent() == null) {
+                    logger.debug("Parent is null for tumor type code '" +
+                        tumorType.getCode() + "'.  Adding to 'TISSUE' node.");
+                    tumorType.setParent(root.getCode());
+                    root.addChild(tumorType);
```

Comment:
right .. constructing all parent and child relationships based on the parent. This makes sense.